testing/tests/redirects.test.js

```diff
@@ -0,0 +1,1034 @@
+  // url_test("/css/wiki.css", null, { statusCode: 404 }),
+  // url_test("/css/base.css", null, { statusCode: 404 }),
+  url_test("/contests", "http://www.mozillalabs.com/", { statusCode: 302 }),
+  url_test("/contests/", "http://www.mozillalabs.com/", { statusCode: 302 }),
+  url_test("/contests/extendfirefox/faq.php", "http://www.mozillalabs.com/", {
+    statusCode: 302,
+  }),
+  url_test("/es4", "http://www.ecma-international.org/memento/TC39.htm", {
+    statusCode: 302,
+  }),
+  url_test("/es4/", "http://www.ecma-international.org/memento/TC39.htm", {
+    statusCode: 302,
+  }),
+  url_test(
+    "/es4/proposals/slice_syntax.html",
+    "http://www.ecma-international.org/memento/TC39.htm",
+    { statusCode: 302 }
+  )
+  // TODO: implement locale redirects
+  //url_test(
+  //  "/en/docs/Web/CSS/Attribute_selectors",
+  //  "/en-US/docs/Web/CSS/Attribute_selectors",
+  //  { statusCode: 302 },
+  //),
+  //url_test(
+  //  "/en/docs/Web/CSS/Attribute_selectors",
+  //  "/en-US/docs/Web/CSS/Attribute_selectors",
+  //  { statusCode: 302 },
+  //),
+  //url_test("/cn/docs/Talk:Kakurady", "/zh-CN/docs/Talk:Kakurady", {
+  //  statusCode: 302,
+  //}),
+  //url_test(
+  //  "/zh_cn/docs/Web/API/RTCPeerConnection/addTrack",
+  //  "/zh-CN/docs/Web/API/RTCPeerConnection/addTrack",
+  //  { statusCode: 302 },
+  //),
+  //url_test("/zh_tw/docs/AJAX", "/zh-TW/docs/AJAX", { statusCode: 302 }),
+);
+
+const ZONE_REDIRECTS = [
+  [
+    "Add-ons",
+    "Mozilla/Add-ons",
+    "WebExtensions",
+    [
+      "ar",
+      "bn",
+      "ca",
+      "de",
+      "en-US",
+      "es",
+      "fa",
+      "fr",
+      "hu",
+      "id",
+      "it",
+      "ja",
+      "ms",
+      "nl",
+      "pl",
+      "pt-BR",
+      "pt-PT",
+      "ru",
+      "sv-SE",
+      "th",
+      "uk",
+      "vi",
+      "zh-CN",
+      "zh-TW",
+      null,
+    ],
+  ],
+  ["Add-ons", "Mozilla/Πρόσθετα", "WebExtensions", ["el"]],
+  ["Add-ons", "Mozilla/애드온들", "WebExtensions", ["ko"]],
+  ["Add-ons", "Mozilla/Eklentiler", "WebExtensions", ["tr"]],
+  [
+    "Firefox",
+    "Mozilla/Firefox",
+    "Privacy",
+    [
+      "ar",
+      "bm",
+      "ca",
+      "de",
+      "el",
+      "en-US",
+      "es",
+      "fi",
+      "fr",
+      "he",
+      "hi-IN",
+      "hu",
+      "id",
+      "it",
+      "ja",
+      "ko",
+      "ms",
+      "my",
+      "nl",
+      "pl",
+      "pt-BR",
+      "pt-PT",
+      "ru",
+      "sv-SE",
+      "th",
+      "tr",
+      "vi",
+      "zh-CN",
+      "zh-TW",
+      null,
+    ],
+  ],
+  ["Firefox", "Mozilla/ফায়ারফক্স", "Privacy", ["bn"]],
+  [
+    "Apps",
+    "Web/Apps",
+    "Tutorials",
+    ["en-US", "fa", "fr", "ja", "th", "zh-CN", "zh-TW", null],
+  ],
+  ["Apps", "Web/Aplicaciones", "Tutorials", ["es"]],
+  ["Apps", "Apps", "Tutorials", ["bn", "de", "it", "ko", "pt-BR", "ru"]],
+  // TODO: inconsistent also in SCL3
+  //["Learn", "Learn", "JavaScript", ["ca", "de", null]],
```

Comment:
@escattone that has overlapping paths with https://github.com/mdn/yari/pull/1359/files#diff-0b1bb09219f3058ee09e293b7cd7f930R693 and they differ in 302 vs 301

Reply:
@fiji-flo 👍 I noticed that for the first time last night as well.